src/action/interactive.rs

```diff
@@ -171,15 +177,33 @@ impl UserPicked {
         let KeyEvent { code, modifiers } = event;
 
         match code {
-            KeyCode::Up => state.select_previous(),
-            KeyCode::Down => state.select_next(),
+            KeyCode::Left => {
+                state.i_cursor_index = state.i_cursor_index.checked_sub(1).unwrap_or_default()
+            }
+            KeyCode::Right => {
+                state.i_cursor_index =
+                    (state.i_cursor_index + 1).min(state.custom_replacement.len() as u16)
+            }
+            KeyCode::Up => {
+                state.i_cursor_index = state.custom_replacement.len() as u16;
+                state.select_previous();
+            }
+            KeyCode::Down => {
+                state.i_cursor_index = state.custom_replacement.len() as u16;
+                state.select_next();
+            }
             KeyCode::Enter => {
                 let bandaid = BandAid::new(&state.custom_replacement, &state.suggestion.span);
                 return Ok(Pick::Replacement(bandaid));
             }
             KeyCode::Esc => return Ok(Pick::Quit),
             KeyCode::Char('c') if modifiers == KeyModifiers::CONTROL => return Ok(Pick::Quit),
-            KeyCode::Char(c) => state.custom_replacement.push(c), // @todo handle cursors and insert / delete mode
+            KeyCode::Char(c) => {
+                state
+                    .custom_replacement
+                    .insert(state.i_cursor_index as usize, c);
+                state.i_cursor_index += 1;
+            } // @todo handle cursors and insert / delete mode
```

Comment:
the `//@todo` is obsolete then, isn't it?

Reply:
50% is, left it as a reminder to `overwriting` mode provided by `Insert` key.

But In any way I forgot to implement delete handle :(
I will do it in the nearest.

Reply:
9c8a0fa

Reply:
I think it can be removed, after all the insert mode is kinda optional :)